utils/s3/client.cc

```diff
@@ -470,13 +470,16 @@ class client::readable_file : public file_impl {
     }
 
     virtual future<struct stat> stat(void) override {
-        auto size = co_await _client->get_object_size(_object_name);
+        auto object_stats = co_await _client->get_object_stats(_object_name);
         struct stat ret {};
         ret.st_nlink = 1;
         ret.st_mode = S_IFREG | S_IRUSR | S_IRGRP | S_IROTH;
-        ret.st_size = size;
+        ret.st_size = object_stats.size;
         ret.st_blksize = 1 << 10; // huh?
-        ret.st_blocks = size >> 9;
+        ret.st_blocks = object_stats.size >> 9;
+        // objects are immutable, therefore both modification and creation time can be set
```

Comment:
do you meant to put "neither modification nor creation time can be set"?

Reply:
No. The reasoning is that objects are immutable on S3, therefore we can use Last-Modified to set both st_mtime and st_ctime

Reply:
brw, posix ctime isn't "create" time but rather, it stands for `Time of last status change` (see https://pubs.opengroup.org/onlinepubs/009696799/basedefs/sys/stat.h.html), and it is updated whenever the inode metadata changes (e.g. hardlink or chmod). Anyhow it's perfectly fine to set it equal to mtime.

Reply:
> brw, posix ctime isn't "create" time but rather, it stands for `Time of last status change` (see https://pubs.opengroup.org/onlinepubs/009696799/basedefs/sys/stat.h.html), and it is updated whenever the inode metadata changes (e.g. hardlink or chmod). Anyhow it's perfectly fine to set it equal to mtime.

you're right. one refers to metadata change, whereas the other to actual data change. will update the comment. thanks.

Reply:
> No. The reasoning is that objects are immutable on S3, therefore we can use Last-Modified to set both st_mtime and st_ctime

I see. Could you please use this line in the comment? It helps me to understand the code better than the existing one.